packages/mobile/src/hooks/useNavigation.ts

```diff
@@ -36,20 +36,27 @@ export const useNavigation = () => {
   const nativeNavigation = useNavigationNative<NavigationType>()
   const pushRouteWeb = usePushRouteWeb()
 
-  const navigate = useCallback(
-    <RouteName extends keyof AppParamList>(
+  const performNavigation = useCallback(
+    method => <RouteName extends keyof AppParamList>(
       config: UseNavigationConfig<AppParamList, RouteName>
     ) => {
       const { native, web } = config
-
-      nativeNavigation.navigate(native.screen, native.params)
+      method(native.screen, native.params)
       if (web) {
         pushRouteWeb(web.route, web.fromPage, web.fromNativeNotifications)
       }
     },
-    [nativeNavigation, pushRouteWeb]
+    [pushRouteWeb]
   )
 
-  const navigation = useMemo(() => ({ navigate }), [navigate])
-  return navigation
+  return useMemo(
+    () => ({
+      navigate: performNavigation(nativeNavigation.navigate),
+      push: performNavigation(nativeNavigation.push),
+      // Notifying the web layer of the pop action
+      // is handled in `createStackScreen`
+      goBack: nativeNavigation.goBack
```

Comment:
mind giving a lil context on why we dont just do goBackWeb here?

Reply:
We aren't using `useNavigation` everywhere. When clicking the back button for example, `goBack` is called on the navigation prop. Using an event handler means that any pop will notify the web layer, not just pops using `useNavigation`